packages/wasm/crate/src/build.rs

```diff
@@ -1,28 +1,27 @@
-use crate::error::WasmResult;
-use std::str::FromStr;
-use wasm_bindgen::prelude::wasm_bindgen;
-use wasm_bindgen::JsValue;
-
-use crate::utils;
 use penumbra_keys::FullViewingKey;
 use penumbra_transaction::{
     plan::{ActionPlan, TransactionPlan},
     WitnessData,
 };
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+
+use crate::error::WasmResult;
+use crate::utils;
 
 /// Builds a planned [`Action`] specified by
 /// the [`ActionPlan`] in a [`TransactionPlan`].
 /// Arguments:
 ///     transaction_plan: `TransactionPlan`
 ///     action_plan: `ActionPlan`
-///     full_viewing_key: `bech32m String`,
+///     full_viewing_key: `Full`,
 ///     witness_data: `WitnessData``
 /// Returns: `Action`
 #[wasm_bindgen]
 pub fn build_action(
     transaction_plan: JsValue,
     action_plan: JsValue,
-    full_viewing_key: &str,
+    full_viewing_key: JsValue,
```

Comment:
Somewhat related to https://github.com/penumbra-zone/web/pull/844, but we can do something like this so we get types:

```rust
pub fn build_action(
    transaction_plan: JsValue,
    action_plan: JsValue,
    full_viewing_key: &[u8],
    witness_data: JsValue,
) -> WasmResult<JsValue> {
    let full_viewing_key: FullViewingKey = FullViewingKey::decode(full_viewing_key)?;
```

and then in typescript we pass `fullViewingKey.inner`

Reply:
Why is it better? Do we want to get rid of `toJson()` serialization and `serde_wasm_bindgen`? 
Should we do this in all cases when handling `{inner: Uint8Array}` types?

Reply:
I'm working through these cases in https://github.com/penumbra-zone/web/pull/834. I go back and forth honestly. Perhaps it's not better than JsValue. We still pass an opaque type that needs deserializing on the other side. Maybe ignore this for now.

Reply:
Ok, here's the idea with this:
- wasm: https://github.com/penumbra-zone/web/pull/834/files#diff-8379276b019dd49da4c1b561332a55996d838efd25a8e02609f6ed40625e94d1R103-R108
- typescript: https://github.com/penumbra-zone/web/pull/834/files#diff-be70ea133ec02a72fd2fc4cd0b3c2782d9ce8d074a426f5490e9abf78e49c29bR14-R17

`JsValue` can be _anything_. And it gets typed to an `any`. However, in some cases, we can encode/decode through binary. If we pass an Uint8Array straight to wasm and receive one back, the typescript signature will be strongly enforced. It increases typesafety. It won't work in every case, but for pb types, it often can.